src/docker-osx-dev

```diff
@@ -371,14 +371,13 @@ function init_boot2docker {
     boot2docker init
   fi
 
-  check_for_shared_folders
-
   if ! is_boot2docker_running; then
     log_info "Starting Boot2Docker VM"
     boot2docker start --vbox-share=disable
   fi
 
   configure_boot2docker
+  check_for_shared_folders
```

Comment:
Shared folder detection/removal currently uses `boot2docker ssh {mount,umount}`, so I think it will actually error if _isn't_ run after starting the instance. Previously, it looks like it was using `VBoxManage`, so it probably was the other way around then.